synapseclient/core/utils.py

```diff
@@ -242,6 +250,38 @@ def id_of(obj: typing.Union[str, collections.abc.Mapping, numbers.Number]) -> st
     raise ValueError("Invalid parameters: couldn't find id of " + str(obj))
 
 
+def validate_submission_id(
+    submission_id: typing.Union[str, int, collections.abc.Mapping]
+) -> str:
+    """
+    Ensures that a given submission ID is either an integer or a string that
+    can be converted to an integer. Version notation is not supported for submission
+    IDs, therefore decimals are not allowed.
+
+    Arguments:
+        submission_id: The submission ID to validate
+
+    Returns:
+        The submission ID as a string
+
+    """
+    if isinstance(submission_id, int):
+        return str(submission_id)
+    elif isinstance(submission_id, str) and submission_id.isdigit():
+        return submission_id
+    elif isinstance(submission_id, collections.abc.Mapping):
+        syn_id = _get_from_members_items_or_properties(submission_id, "id")
+        if syn_id is not None:
+            return validate_submission_id(syn_id)
+    else:
+        int_submission_id = int(float(submission_id))
```

Comment:
What is the error if we pass in a random string?  Currently it's.
```
SynapseHTTPError: 404 Client Error: 
Submission 'fasdf' does not exist
```

I think the error will become:
```
----> 1 float("fasdf")

ValueError: could not convert string to float: 'fasdf'
```

which _could_ be fine

Reply:
Good point, I've implemented a catch for non-digit values in a string input:

```
In [55]: sub = syn.getSubmission("test")
---------------------------------------------------------------------------
[...]
ValueError: could not convert string to float: 'test'

During handling of the above exception, another exception occurred:

[...]

ValueError: Submission ID 'test' is not a valid submission ID. Please use digits only.
```